src/plugins/data_view_management/public/management_app/data_view_management_service.ts

```diff
@@ -0,0 +1,293 @@
+}
+
+export type BehaviorObservable<T> = Omit<BehaviorSubject<T>, 'next'>;
+
+export const matchedIndiciesDefault = {
+  allIndices: [],
+  exactMatchedIndices: [],
+  partialMatchedIndices: [],
+  visibleIndices: [],
+};
+
+/**
+ * ConstructorArgs for DataViewEditorService
+ */
+export interface DataViewMgmtServiceConstructorArgs {
+  /**
+   * Dependencies for the DataViewEditorService
+   */
+  services: {
+    application: ApplicationStart;
+    dataViews: DataViewsPublicPluginStart;
+    savedObjectsManagement: SavedObjectsManagementPluginStart;
+    uiSettings: IUiSettingsClient;
+  };
+  /**
+   * Whether service requires requireTimestampField
+   */
+  requireTimestampField?: boolean;
+  /**
+   * Initial type, indexPattern, and name to populate service
+   */
+  initialValues: {
+    name?: string;
+    type?: INDEX_PATTERN_TYPE;
+    indexPattern?: string;
+  };
+}
+
+export interface DataViewMgmtState {
+  dataView?: DataView;
+  allowedTypes: SavedObjectManagementTypeInfo[];
+  relationships: SavedObjectRelationWithTitle[];
+  fields: DataViewField[];
+  scriptedFields: DataViewField[];
+  scriptedFieldLangs: Array<{
+    value: string;
+    name: string;
+    checked?: FilterChecked;
+  }>;
+  indexedFieldTypes: string[];
+  fieldConflictCount: number;
+  tags: Array<{ key: string; 'data-test-subj': string; name: string }>;
+  isRefreshing: boolean;
+  defaultIndex: string;
+  conflictFieldsUrl: string;
+}
+
+const defaultDataViewEditorState: DataViewMgmtState = {
+  allowedTypes: [],
+  relationships: [],
+  fields: [],
+  scriptedFields: [],
+  scriptedFieldLangs: [],
+  indexedFieldTypes: [],
+  fieldConflictCount: 0,
+  tags: [],
+  isRefreshing: true,
+  defaultIndex: '',
+  conflictFieldsUrl: '',
+};
+
+export const stateSelectorFactory =
+  <S>(state$: Observable<S>) =>
+  <R>(selector: (state: S) => R, equalityFn?: (arg0: R, arg1: R) => boolean) =>
+    state$.pipe(map(selector), distinctUntilChanged(equalityFn));
+
+export class DataViewMgmtService {
+  constructor({
+    services: { dataViews, savedObjectsManagement, uiSettings, application },
+    initialValues: {},
+  }: DataViewMgmtServiceConstructorArgs) {
+    this.services = {
+      application,
+      dataViews,
+      savedObjectsManagement,
+      uiSettings,
+    };
+
+    this.internalState$ = new BehaviorSubject<DataViewMgmtState>({
+      ...defaultDataViewEditorState,
+    });
+
+    this.state$ = this.internalState$ as BehaviorObservable<DataViewMgmtState>;
+
+    // allowed types are set once and never change
+    this.allowedTypes = new Promise((resolve) => {
+      savedObjectsManagement.getAllowedTypes().then((resp) => {
+        resolve(resp);
+      });
+    });
+  }
+
+  private services: {
+    application: ApplicationStart;
+    dataViews: DataViewsPublicPluginStart;
+    savedObjectsManagement: SavedObjectsManagementPluginStart;
+    uiSettings: IUiSettingsClient;
+  };
+
+  private allowedTypes: Promise<SavedObjectManagementTypeInfo[]>;
```

Comment:
Should not it be of `SavedObjectManagementTypeInfo[]` type?

Reply:
Its using a promise to avoid potential race conditions.